src/config/groups/keys.cpp

```diff
@@ -1432,6 +1432,14 @@ LIBSESSION_C_API int groups_keys_init(
     return SESSION_ERR_NONE;
 }
 
+LIBSESSION_C_API void groups_keys_free(config_group_keys* conf) {
+    delete conf;
```

Comment:
Is it enough to just delete the `conf`? `conf->internals` is a raw pointer it seems.

Reply:
I just replicated the existing `config_free` function 😬 
```
LIBSESSION_EXPORT void config_free(config_object* conf) {
    delete conf;
}
```

Reply:
Hmm I think you do need to delete the internals here because
1. `config_object`/`config_groupb_keys` is a plain struct (it doesn't have a destructor)
2. `internal` is created using `new` and put into a raw pointer, that means it doesn't have a smart pointer to do the clean up.

Unless I miss something I think this one will leak memory (so does the config one)

Reply:
They are all copied from an original incomplete C destruction that does indeed leak.  I've pushed a commit here to fix it (both here, and two other places).  There's another one (for the network object) in the libquic API that I'll push there.

Reply:
Thanks for pushing up a fix for these, and good catch @simophin 🙏